### ISSUE TITLE
The `path.data` option appears not to be respected in configuration files

### ISSUE THREAD

**Issue by cachedout** [elastic/beats#25143]
## Version

[root@centos etc]# metricbeat version
metricbeat version 7.12.0 (amd64), libbeat 7.12.0 [08e20483a651ea5ad60115f68ff0e53e6360573a built 2021-03-18 06:28:33 +0000 UTC]

## Description

It appears to me that setting `path.data` via a configuration file is not working. I had originally tested this in the context of trying to get multiple Beats to run on a single host, but the problem seems to occur even in a single-Beat scenario.

## Reproduction

Consider the following configuration file:

```
[root@centos etc]# cat /etc/metricbeat-monitor/metricbeat-monitor.yml

name: metribeat-monitor
tags: ['monitor']
index: metricbeat-monitor
path.data: /my_metricbeat_path
path.logs: /var/log/metricbeat-monitor
```

This configuration file is the only file present in the directory:

```
[root@centos etc]# ls -l /etc/metricbeat-monitor/
total 4
-rw-------. 1 root root 633 Apr 19 15:35 metricbeat-monitor.yml
```

I start Metricbeat using this configuration:

```
metricbeat -e -v  --path.config /etc/metricbeat-monitor -c metricbeat-monitor.yml
```

After letting Metricbeat run for a while (~30 seconds), I observe that the `/my_metricbeat_path` directory remains empty:

```
[root@centos etc]# ls -l /my_metricbeat_path/
total 0
```

I then restart Metricbeat, but this time, I add `--path-data /my_metricbeat_path` to the flags when starting:

```
metricbeat -e -v  --path.config /etc/metricbeat-monitor --path.data /my_metricbeat_path -c metricbeat-monitor.yml
```

I let Metricbeat again run for a few moments (< 30s) and then again observe the directory:

```
[root@centos metricbeat]# ls -l /my_metricbeat_path/
total 4
-rw-------. 1 root root 48 Apr 19 15:51 meta.json
```

## Impact

This seems to break documented behavior in at least a few places:

https://www.elastic.co/guide/en/beats/metricbeat/7.12/configuration-path.html#_data

https://www.elastic.co/guide/en/beats/libbeat/7.9/breaking-changes-7.6.html#_two_beat_instances_can_no_longer_share_the_same_data_path

https://www.elastic.co/guide/en/beats/metricbeat/7.12/monitoring-metricbeat-collection.html#configure-metricbeat

## Other versions

I have also tested this in 7.9.0 and observed the same change in behavior.

**Comment by cachedout** [comment]
I think I found the problem here. This was installed via a RPM and `/bin/metricbeat` is not a binary, but in fact is a shell script. This shell script is as follows:

```
#!/usr/bin/env bash

# Script to run Metricbeat in foreground with the same path settings that
# the init script / systemd unit file would do.

exec /usr/share/metricbeat/bin/metricbeat \
  --path.home /usr/share/metricbeat \
  --path.config /etc/metricbeat \
  --path.data /var/lib/metricbeat \
  --path.logs /var/log/metricbeat \
  "$@"
```

This means that when you call `/bin/metricbeat` and pass in `--path.config` and a `-c` option to indicate a custom config, what you end up with is a process that looks like this:

`/usr/share/metricbeat/bin/metricbeat --path.home /usr/share/metricbeat --path.config /etc/metricbeat --path.data /var/lib/metricbeat --path.logs /var/log/metricbeat -e --path.config /etc/metricbeat-monitor --path.data /var/lib/metricbeat -c metricbeat-monitor.yml`

This has the effect whereby configuration values in the hard-coded configuration file take precedence over values specified in your custom config file.

The tl;dr here is that custom configuration values **do not work**  in Metricbeat. If you specify a custom config value via a configuration file, that value will be overridden by the same value from the default configuration.

**Comment by cachedout** [comment]
@faec Eh? I am not sure this is solved. I haven't tested it lately, but AFAIK the last paragraph of my latest post still applies:

```
The tl;dr here is that custom configuration values in config files do not work in Metricbeat when installed via RPM. If you specify a custom config value via a configuration file, that value will be overridden by the same value from the default configuration.
```
